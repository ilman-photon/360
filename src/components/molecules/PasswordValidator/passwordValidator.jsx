// import * as React from "react";
import React, { useEffect, useRef } from "react";
import { LabelWithIcon } from "../../atoms/LabelWithIcon/labelWithIcon";
import { useForm } from "react-hook-form";

export const PasswordValidator = ({ ...props }) => {
    const validator = props.validator || []
    let isValid = true
    if (props.password) {
        //validate
        validator.forEach(element => {
            if (element.validate) isValid = false
        })
    }

    return (
        <div style={{ display: "block" }} >
            {props.isShowValidation ?
                <>
                    {validator.map((err,i) => {
                        return <LabelWithIcon key={i} error={err.validate} label={err.label} />
                    })}
                </> : null}
        </div >
    );
}

export default PasswordValidator