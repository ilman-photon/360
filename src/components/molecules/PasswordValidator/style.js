export const styles = {
textStyles: {
    fontWeight: "400",
    fontSize: "1rem",
    lineHeight: "18px",
    color: "#191919",
    margin: "0",
    marginTop: "15px"
  },
}