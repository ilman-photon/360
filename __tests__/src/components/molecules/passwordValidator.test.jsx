import { render, screen } from "@testing-library/react";

import PasswordValidator from "../../../../src/components/molecules/PasswordValidator/passwordValidator";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

describe("StyledInput", () => {
  it("renders input components with type text", async () => {
    const passwordValidator = [
      { label: "Contain at least 3 our of 4 types", text: true },
    ];
    const { container } = render(
      <PasswordValidator
        isShowValidation={true}
        validator={passwordValidator}
        password="password"
        validatePassword={jest.fn()}
      />
    );
    // expect(
    //   await screen.getByRole("heading", {
    //     name: "Contain at least 3 our of 4 types",
    //   })
    // ).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("renders input components with type text", async () => {
    const passwordValidator = [
      {
        label: "Password length should range from 8 to 20 characters",
        validate: !RegExp(/^[^-\s]{8,20}$/),
      },
    ];
    const { container } = render(
      <PasswordValidator
        isShowValidation={true}
        validator={passwordValidator}
        password="password"
        validatePassword={jest.fn()}
      />
    );
    expect(container.textContent).toBe(
      "Password length should range from 8 to 20 characters"
    );
  });

  it("renders input components with type text", async () => {
    const passwordValidator = [
      {
        label: "Password should contain at least one numerical character (0-9)",
        validate: new RegExp("[0-9]"),
      },
    ];
    const { container } = render(
      <PasswordValidator
        isShowValidation={true}
        validator={passwordValidator}
        password="password"
        validatePassword={jest.fn()}
      />
    );
    // const input = container.getElementsByClassName("custom-input");
    // expect(input).toBeTruthy();
    expect(container.textContent).toBe(
      "Password should contain at least one numerical character (0-9)"
    );
    // expect(await screen.getByRole('span', { name: 'Contain at least 3 our of 4 types'})).toBeVisible()
    // expect(await screen.findByText('Contain at least 3 our of 4 types')).toBeVisible();
    // const button = screen.getByText("Password should contain at least one numerical character (0-9)").closest();
    // debug(button);
    // const node = screen.getByText("Reset");
    // expect(button.closest("button")).toHaveProperty("error", "true");
    // expect(container.firstChild).toMatchSnapshot()
    // expect(container).toMatchSnapshot();
  });
});
