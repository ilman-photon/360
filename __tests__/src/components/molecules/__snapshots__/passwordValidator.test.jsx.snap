// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput renders input components with type text 1`] = `
<div>
  <div
    style="display: block; margin: 8px; padding-bottom: 8px;"
  >
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiBox-root css-0"
            style="display: flex; flex-direction: row; margin-top: 15px;"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ppdlaf-MuiSvgIcon-root"
              data-testid="CheckCircleRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.29 16.29 5.7 12.7a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l6.88-6.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-7.59 7.59c-.38.39-1.02.39-1.41 0z"
              />
            </svg>
            <span
              style="margin-left: 10px; color: rgb(22, 136, 69); font-family: Libre Franklin; font-style: normal; font-weight: 500; font-size: 14px; line-height: 18px;"
            >
              Contain at least 3 our of 4 types
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
